src/main/org/codehaus/groovy/runtime/typehandling/ShortTypeHandling.java

```diff
@@ -41,7 +43,15 @@ public static Class castToClass(Object object) {
 
     public static String castToString(Object object) {
         if (object==null) return null;
-        if (object instanceof Class) return (String) object;
+        if (object instanceof boolean[]) return Arrays.toString((boolean[])object);
```

Comment:
Would it be more efficient to put this within an `if(object.getClass().isArray)` block?